docs/reference/plugins.md

```diff
@@ -2,22 +2,252 @@
+Sie können für die Gerätekategorien [`meter`](/docs/reference/configuration/meters#custom) (Strommessgeräte), [`charger`](/docs/reference/configuration/chargers#type) (Wallboxen) oder [`vehicle`](/docs/devices/vehicles#manuell) (Fahrzeuge) verwendet werden.
+Plugins können auch für die in [Messaging](/docs/reference/configuration/messaging) beschriebenen Endpunkte zum Versenden von Lifecycle-Events genutzt werden.
+
+Je nach Verwendung werden Plugins **lesend** oder **schreibend** eingesetzt.
+
+## Übersicht
+
+Folgende Plugins können verwendet werden, um externe Datenquellen einzubinden:
+
+- [Modbus Plugin](#modbus) - Plugin zum Auslesen von einem Modbus-fähigen Gerät.
+- [MQTT Plugin](#mqtt) - Plugin um indirekt über MQTT mit den MQTT-fähigen Geräten zu kommunizieren.
+- [HTTP Plugin](#http) - Plugin, das über HTTP-API mit Endgeräten spricht.
+- [Websocket Plugin](#websocket) - Plugin zum Empfangen von Gerätedaten über einen eigenen Webserver. Kann nur zum Lesen von Daten genutzt werden.
+- [SMA/Speedwire Plugin](#speedwire) - Plugin speziell für SMA Geräte, die mit dem Speedwire Protokoll kommunizieren können.
+- [JavaScript Plugin](#javascript) - Plugin, das Werte in über ein JavaScript Skript bereitstellt oder entgegennimmt.
+- [Shell Plugin](#shell) - Plugin, das ein Shell Skript ausführen kann, um Daten zu extrahieren oder schreibend entgegennimmt.
+
+Neben diesen Plugins, die externe Daten integrieren, gibt es folgende Helfer-Plugins, die Daten direkt bereitstellen können. Diese können nur in einem lesenden Kontext genutzt werden:
+
+- [Const Plugin](#const) - Spezielles Plugin das einfach einen konstanten Wert zurückliefert.
+- [Calc Plugin](#calc) - Meta-Plugin um Ausgaben von anderen Plugins arithmetisch zu verknüpfen.
+- [Combined Plugin](#combined) - Meta-Plugin speziell für `charger` um die booleschen Status-Werte für den angeschlossenen (_plugged_) und ladenden (_charging_) Zustand zu einem einzigen Ladestatus zu kombinieren.
+
+### Syntax
+
+Jedes Plugin besitzt ein individuelles Konfigurationsschema.
+Dabei ist es wichtig zu wissen, ob das Plugin in einem **lesenden** oder **schreibenden** Kontext verwendet wird.
+Einige Konfigurationsparameter machen nur in einem lesenden Kontext Sinn, andere nur, wenn sie im Schreibmodus genutzt werden.
+Die meisten Konfigurationsparameter sind Plugin spezifisch, jedoch gibt es eine handvoll Parameter, die beim Lesen von einem Plugin bzw. beim Schreiben via eines Plugins generell genutzt werden können.
+
+Beispielsweise kann über die folgende Konfiguration ein MQTT Plugin als `meter` eingebunden werden, bei dem der aktuelle Stromverbrauch über das spezifizierte MQTT Topic eingelesen wird:
+
+```yaml title="Beispiel: MQTT Plugin für die Leistungswerte eines Strommessgeräts"
+meters:
+  - name: imsys
+    type: custom
+    power:
+      source: mqtt
+      topic: "home/current/imsys/chn2/raw"
+```
+
+Das Schema hat dabei immer folgende Struktur:
+
+```yaml {3,5-6,8}
```

Comment:
@naltatis Hast Du eine Idee, wie ich die Platzhalter (`<...>`) hervorheben kann ? Ich habs mit Prism plugins probiert, aber React/Docsaurus-Fu ist nicht gut genug. Idealerweise waeren die Platzhale in kursiv + bold. 

Ist aber nicht dringend, man kann sicher auch ohne leben. Nur falls Du zufaellig was weisst.

Testweise habe ich line-highlight Prism plugin enabled, aber so toll ist das hier auch nicht (sollte man wohl auch mit `magic comments`)

Reply:
[Line-Highlights](https://docusaurus.io/docs/markdown-features/code-blocks#line-highlighting) mit Magic Comments sollten ja ohne Konfigurationsänderung schön möglich sein:

```yaml
tariffs:
  currency: EUR # (default EUR)
  grid:
    // highlight-start
    type: fixed
    // highlight-end
    price: 0.294 # EUR/kWh
```

Aber du möchtest nur die Teilwörter, keine Zeilen richtig? Da bin ich auch überfragt.